store/web3.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  VuexAction as Action,
+  VuexMutation as Mutation,
+  VuexModule,
+  Module,
+} from "nuxt-property-decorator"
+import WalletConnectProvider from "@walletconnect/web3-provider"
+import Web3modal from "web3modal"
+import { ethers } from "ethers"
+
+const providerOptions = {
+  walletconnect: {
+    package: WalletConnectProvider,
+    options: { infuraId: "3ce35c3d389a4461bffd073fbf27d23e" },
+  },
+}
+
+let web3Modal: Web3modal
+let currentProvider: ethers.providers.Web3Provider
```

Comment:
Why is the current Provider not part of the store module ?
This way I am not sure if vuex will detect changes etc. the way you would want it to.

Reply:
Looks like the most recommended way is to keep the instance outside of the store. 
example:
https://github.com/coatofbits/vuex-web3

this one uses a function but ultimately is attaching the instance to the window
https://github.com/jcbombardelli/vuex-web3